frontend/components/Cell.js

```diff
@@ -67,150 +67,191 @@ export const empty_cell_data = (cell_id) => {
  */
 export const code_differs = (cell) => cell.remote_code.body !== cell.local_code.body
 
-export const Cell = ({
-    cell_id,
-    remote_code,
-    local_code,
-    code_folded,
-    queued,
-    running,
-    runtime,
-    errored,
-    output,
-    selected,
-    on_change,
-    on_update_doc_query,
-    on_focus_neighbor,
-    disable_input,
-    focus_after_creation,
-    scroll_into_view_after_creation,
-    all_completed_promise,
-    selected_friends,
-    requests,
-    client,
-    notebook_id,
-}) => {
-    // cm_forced_focus is null, except when a line needs to be highlighted because it is part of a stack trace
-    const [cm_forced_focus, set_cm_forced_focus] = useState(null)
-
-    useEffect(() => {
-        const focusListener = (e) => {
-            if (e.detail.cell_id === cell_id) {
-                if (e.detail.line != null) {
-                    const ch = e.detail.ch
-                    if (ch == null) {
-                        set_cm_forced_focus([{ line: e.detail.line, ch: 0 }, { line: e.detail.line, ch: Infinity }, { scroll: true }])
-                    } else {
-                        set_cm_forced_focus([{ line: e.detail.line, ch: ch }, { line: e.detail.line, ch: ch }, { scroll: true }])
+export class Cell extends Component {
+    constructor () {
+        super()
+    }
+
+    shouldComponentUpdate (next_props) {
+        const comparable_prop_keys = [
```

Comment:
Can we use `Object.keys` or one of its friends?

Reply:
I didn't want to compare all props, because some props are functions which will automatically be different on any change and the component will rerender :/

But I can use Object.keys, and automatically skip props with `typeof prop === 'function'`.

Reply:
Different functions should be memoized with `useCallback` and be `===` same across re-renders.

Reply:
Thanks for the suggestion @pankgeorg! @novoselrok implement that if you think it's appropriate, otherwise just copy your explanation as a comment into the code and its fine